### PR TITLE
Bump phpSPO to 2.4, brings 2019 support

### PR DESCRIPTION
**Pull request by blizzz**
solves #49

* [x] bring back (fall-back to) NTLM auth

### CODE REVIEW

File: lib/Client.php [hunk 1]
```diff
@@ -129,19 +125,14 @@ private function findSharePointErrorCode(Exception $e): ?int {
 		if ($trace['function'] === 'validateResponse' && isset($trace['args'][0])) {
 			$responseCodeJson = json_decode($trace['args'][0], true)['error'];
 		} else {
-			$lastRequest = $this->getLastRequestData();
-			if (!isset($lastRequest['response'])) {
+			if (!isset($this->lastResponse)) {
```
Comment (come-nc) [inline lib/Client.php:128]: redundant with `if (!isset($this->lastResponse['error'])) {` on line 132

File: lib/Client.php [hunk 2]
```diff
@@ -24,20 +24,21 @@
 namespace OCA\SharePoint;
 
 use Exception;
+use OCP\ILogger;
```
Comment (come-nc) [inline lib/Client.php:27]: It seems ILogger was not used before and I don’t see it used in the diff?

**Review by come-nc — APPROVED** [review]

File: lib/Client.php (hunk 2 above)
Comment (blizzz) [reply lib/Client.php:27]: In `ensureConnection()`, but let me replace that as well.

**Pull request merged**